### PR TITLE
Phase2-hgx87 Add functionality in geometry to enable avoid magic # usage in reconstruction

### PR DESCRIPTION
**Pull request by bsunanda**
Only equipped RecTools with some of the stuff - to be propagated to other classes in HGCalRecAlgos

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
please test

**Comment by kpedro88** [comment]
@bsunanda thanks
@clelange FYI

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h [hunk 1]
```diff
@@ -73,6 +73,9 @@ class HGCalDDDConstants {
 					int& wafer, int& icell, 
 					int& celltyp) const;
   bool                waferInLayer(int wafer, int lay, bool reco) const;
+  int                 waferKount(const int type) const {return ((type == 0) ? waferMax_[2] : waferMax_[3]);}
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/interface/HGCalDDDConstants.h]: misspelling: Kount -> Count

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 2]
```diff
@@ -64,10 +59,19 @@ void RecHitTools::getEvent(const edm::Event& ev) {
...
   es.get<CaloGeometryRecord>().get(geom);
 
   geom_ = geom.product();
+  auto geomEE = static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE));
+  fhOffset_ = (geomEE->topology().dddConstants()).layers(true);
+  unsigned int wmaxEE = 1 + (geomEE->topology().dddConstants()).waferMax();
+  auto geomFH = static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCHEF));
+  bhOffset_ = fhOffset_ + (geomFH->topology().dddConstants()).layers(true);
+  unsigned int wmaxFH = 1 + (geomFH->topology().dddConstants()).waferMax();
+  maxNumberOfWafersPerLayer_ = (wmaxEE > wmaxFH) ? wmaxEE : wmaxFH;
```
Comment (kpedro88) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: could just be `std::max()`

**Comment by clelange** [comment]
alright, thanks, this is useful.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @kpedro88 @slava77 Kindly approve this PR

**Comment by perrotta** [comment]
+1
Reorganization of the HGCal code which allows to retrieve centrally some of the magic numbers popped up in the previous versions
No effect on the workflows tested in jenkins, as expected

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
merge

**Pull request merged**